app/controllers/onboarding_controller.rb

```diff
@@ -17,28 +17,42 @@ def create
     return redirect_to_success if User.email_taken?(user_params[:email])
 
     @user = User.new(user_params)
-    return redirect_to_success if @user.save
+    if @user.save
+      JsonWebToken.create(invalidated_jti: params[:jwt])
+      return redirect_to_success
+    end
 
     render :index
   end
 
   def success; end
 
+  def create_invite_url
+    payload = SecureRandom.base64(16)
+    jwt = JsonWebToken.encode(payload)
+    render json: { url: onboarding_url(jwt: jwt) }
+  end
+
   private
 
   def redirect_to_success
-    redirect_to onboarding_success_path(token: token_param)
+    redirect_to onboarding_success_path(jwt: jwt_param)
```

Comment:
This doesn’t actually work. When the user is redirected to the success page, the token is already invalidated. I’d suggest to simply allow visiting the `/onboarding/success` page without a token as it doesn’t contain any sensitive information.

Reply:
I agree... maybe I should have asked, but I wasn't sure why we had the token there in the first place.